hathor/p2p/sync_v2/agent.py

```diff
@@ -407,10 +392,12 @@ def run_sync_blocks(self) -> Generator[Any, Any, bool]:
                        synced_block=self.synced_block)
 
         # Sync from common block
-        yield self.start_blockchain_streaming(self.synced_block.id,
-                                              self.synced_block.height,
-                                              self.peer_best_block.id,
-                                              self.peer_best_block.height)
+        try:
+            yield self.start_blockchain_streaming(self.synced_block,
+                                                  self.peer_best_block)
+        except StreamingError:
+            self.send_stop_block_streaming()
```

Comment:
Should we do anything else here? Maybe set state to ERROR? Or just close connection?

Reply:
I improved it in PR #850 .